eth_crowdfund_api/src/views/RequestView.py

```diff
@@ -42,25 +38,37 @@ def get_all_requests():
 @request_api.route('/<int:request_id>', methods=['GET'])
 def get_a_request(request_id):
   campaign_request = Request.get_one_request(request_id)
-  if not campaign_request:
+  if campaign_request is None:
     return custom_response({ 'error': 'Request not found'}, 404)
   
   campaign_request_data = request_schema.dump(campaign_request)
   return custom_response(campaign_request_data, 200)
 
 @request_api.route('/<int:request_id>', methods=['PUT'])
 def update(request_id):
-  req_data = request.get_json()
-  data = request_schema.load(req_data, partial=True)
 
-  campaign_request = Request.get_one_request(request_id)
-  campaign_request.update(data)
+  try: 
+    req_data = request.get_json()
+    data = request_schema.load(req_data, partial=True)
+    campaign_request = Request.get_one_request(request_id)
+    campaign_request.update(data)
+  except ValidationError as err:
+    return custom_response(err.messages, 400)
+  except exc.IntegrityError as err: 
+    return custom_response({ "error": err.orig.diag.message_detail }, 400)
```

Comment:
This catches errors such as attempting to change campaign_id to an invalid value